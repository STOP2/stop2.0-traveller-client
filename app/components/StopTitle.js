import React, { Component } from 'react'
import { DefaultText } from '../components/textComponents'

import styles from '../styles/stylesheet'

class StopTitle extends Component {
    render()
    {
        return (<DefaultText style={styles.stopTitle}>
                  {this.props.name} ({this.props.line}) â€” {this.props.distance}m
                </DefaultText>)
    }
}

StopTitle.propTypes = {
    name: React.PropTypes.string.isRequired,
    line: React.PropTypes.string.isRequired,
    distance: React.PropTypes.number.isRequired
}

export default StopTitle
