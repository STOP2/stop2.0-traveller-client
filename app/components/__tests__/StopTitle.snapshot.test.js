import 'react-native'
import React from 'react'
import StopTitle from '../StopTitle'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'


jest.mock('react-native-localization')

describe('StopTitle component', () =>
{
    it('renders correctly', () =>
    {
        let tree = renderer.create(<StopTitle name="A.I. Virtasen aukio" line="5302" />).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
