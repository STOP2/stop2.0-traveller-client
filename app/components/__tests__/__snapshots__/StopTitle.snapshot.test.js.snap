exports[`StopTitle component renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontFamily": "gotham-rounded-book",
      },
      Object {
        "backgroundColor": "#007AC9",
        "color": "#FFFFFF",
        "fontSize": 20,
        "padding": 5,
      },
    ]
  }>
  A.I. Virtasen aukio
   (
  5302
  )
</Text>
`;
